### ISSUE TITLE
[bug] v2 command + z sometimes doesn't work on macos

### ISSUE THREAD

**Issue by drl990114** [tauri-apps/tauri#9426]
### Describe the bug

I ran into a situation that might be similar.

During the undo process of `command + z` under macos, this does not take effect in some scenarios. For example, if I insert text not through input but through a certain shortcut, `command + z` will not take effect.

This is normal in `tauri.beta.7` and I have temporarily rolled back to that version.

### Reproduction

- input some text 
- command + z can work
- command + c insert some text
- command + z can not work

### Full `tauri info` output

```text
[✔] Environment
    - OS: Mac OS 14.4.0 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.76.0 (07dca489a 2024-02-04)
    ✔ cargo: 1.76.0 (c84b36747 2024-01-18)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 20.11.0
    - pnpm: 8.10.2
    - yarn: 4.1.1
    - npm: 10.2.4
    - bun: 1.1.0

[-] Packages
    - tauri [RUST]: 2.0.0-beta.14
    - tauri-build [RUST]: 2.0.0-beta.11
    - wry [RUST]: 0.39.0
    - tao [RUST]: 0.27.0
    - @tauri-apps/api : not installed!
    - @tauri-apps/cli [NPM]: 2.0.0-beta.9

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:3000/
    - framework: React
    - bundler: Rollup
```

**Comment by pewsheen** [comment]
It seems to work on my side. Could you share a reproducible repo or code to test?

What I did is:
- input something
- command + z (works)
- copy some texts and command+v paste to a text input
- command + z (works)

**Comment by drl990114** [comment]
@pewsheen  I'll try again, and if possible, will provide a reproducible warehouse as soon as possible.

**Comment by drl990114** [comment]
Reproduction repo: https://github.com/drl990114/tauri-issue, `readme,md` provided by reproduction details.

**Comment by pewsheen** [comment]
Looks like https://github.com/tauri-apps/wry/pull/1208 breaks it 😢 

Slate editor listens to the JS `keydown` event and manages redo/undo history itself (from the JS side). The key events from OS are passed through the WKWebView responder and sent to the next responder directly, so the editor didn't receive the `cmd+z` (was consumed by the application menu)

I'll see if there's a solution for it.

cc @lucasfernog

**Comment by lucasfernog** [comment]
in this case isn't the right approach to remove the Undo menu item?

**Comment by pewsheen** [comment]
Natvie input like `<input>` or `contenteditable` elements still require an undo menu item

**Comment by Nordalf** [comment]
Experiencing similar issues with Monaco Editor using `.18` beta version. Only relates to OSX. Windows is working. Just tested.

@lucasfernog

**Comment by Nordalf** [comment]
From Discord where I am running a support thread:

Just to confirm: I forked Wry and Tauri, modified my `cargo.toml` to point at the local repos, and I can confirm everything works as expected after removing line `401-408` in `wry/src/wkwebview/mod/rs`:

```
decl.add_method(
  sel!(performKeyEquivalent:),
  key_equivalent as extern "C" fn(&mut Object, Sel, id) -> BOOL,
);

extern "C" fn key_equivalent(_this: &mut Object, _sel: Sel, _event: id) -> BOOL {
  YES
}
```

**Comment by drl990114** [comment]
This has been fixed in tauri [RUST]: 2.0.0 beta.22.

**Comment by mtsgrd** [comment]
I am working with a lexical.dev editor in a Tauri app, and I am struggling to get the history plugin to work. When I test my components outside of Tauri (safari and chrome) then things work fine, but when loading the same component inside Tauri it can at most undo a few steps and then it stops working.

For what it's worth, it appears as if the number of steps I can undo is equal to the native undo stack for a contenteditable, but the parts that get undone appear to be the undo steps recorded by the lexical editor's history plugin.

I'm a bit unsure how to troubleshoot this, any advice would be appreciated.

*edit: one problem seems to be that KeyZ + metaKey does not actually dispatch on elements inside Tauri, while they do in a regular browser.
